### ISSUE TITLE
Invulnerability applied before teleportAsynch (Syndicate protection/instability)

### ISSUE THREAD

**Issue by MikeJeffers** [OpenPerpetuum/OP-Project#36]
Invulnerability is applied before an synchronous task.
Effect: player client started molecular instability AFTER they load on terrain, protection will count down regardless of when the task completes. Players with >lag have smaller window of protection, with greater amount of instability.

``` C#
public Task DoTeleportAsync(Player player)
    {
      IZone zone = player.Zone;
      if (zone == null)
        return (Task) null;
      Position validPosition;
      if (!new ClosestWalkablePositionFinder(zone, this.TargetPosition, 4.0).Find(out validPosition))
        return (Task) null;
      player.StopAllModules();
      if (this.ApplyInvulnerable) //TODO <<<
        player.ApplyInvulnerableEffect();
      player.States.InMoveable = true;
      player.States.LocalTeleport = true;
      return Task.Delay(this.TeleportDelay).ContinueWith((Action<Task>) (t =>
      {
        player.RemoveFromZone((IBeamBuilder) null);
        player.CurrentSpeed = 0.0;
        player.AddToZone(zone, validPosition, ZoneEnterType.LocalTeleport, (IBeamBuilder) null);
        player.SendInitSelf();
        if (this.ApplyTeleportSickness) //TODO <<<<
          player.ApplyTeleportSicknessEffect();
        player.States.InMoveable = false;
        player.States.LocalTeleport = false;
      }));
    }
```

The fix, re-apply protection on task complete.
(keep durations the same, protection<instability) But address to level field for players with higher ping.

**Comment by MikeJeffers** [comment]
Closed by OpenPerpetuum/PerpetuumServer/@28c72664db0589f5aa33b5a193fdfe3b9e9e8b0a

**Comment by MikeJeffers** [comment]
This is fixed, but the syndicate protection being equal is not good for pvp (but now they have a constant difference ---  this was the issue)
